packages/core/src/routes/hook.test.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  HookEvent,
+  type Hook,
+  type HookEvents,
+  type HookConfig,
+  type CreateHook,
+} from '@logto/schemas';
+import { pickDefault } from '@logto/shared/esm';
+
+import {
+  mockCreatedAtForHook,
+  mockHook,
+  mockHookList,
+  mockNanoIdForHook,
+  mockTenantIdForHook,
+} from '#src/__mocks__/hook.js';
+import { MockTenant } from '#src/test-utils/tenant.js';
+import { createRequester } from '#src/utils/test-utils.js';
+
+const { jest } = import.meta;
+
+const hooks = {
+  findAllHooks: async (): Promise<Hook[]> => mockHookList,
+  insertHook: async (data: CreateHook): Promise<Hook> => ({
+    ...mockHook,
+    ...data,
+  }),
+  findHookById: async (): Promise<Hook> => mockHook,
+  updateHookById: async (_: unknown, data: Partial<CreateHook>): Promise<Hook> => ({
+    ...mockHook,
+    ...data,
+  }),
+  deleteHookById: jest.fn(),
+};
+
+const tenantContext = new MockTenant(undefined, { hooks });
+
+const hookRoutes = await pickDefault(import('./hook.js'));
+
+describe('hook routes', () => {
+  const hookRequest = createRequester({ authedRoutes: hookRoutes, tenantContext });
+
+  it('GET /hooks', async () => {
+    const response = await hookRequest.get('/hooks');
+    expect(response.status).toEqual(200);
+    expect(response.body).toEqual(mockHookList);
+    expect(response.header).not.toHaveProperty('total-number');
+  });
+
+  it('GET /hooks/:id', async () => {
+    const response = await hookRequest.get(`/hooks/${mockNanoIdForHook}`);
+    expect(response.status).toEqual(200);
+    expect(response.body).toEqual(mockHook);
+  });
+
+  it('POST /hooks', async () => {
+    const name = 'fooName';
+    const events: HookEvents = [HookEvent.PostRegister];
+    const config: HookConfig = {
+      url: 'https://example.com',
+    };
+
+    const response = await hookRequest.post('/hooks').send({ name, events, config });
+    expect(response.status).toEqual(200);
+    const generatedId = response.body.id as string;
+    const generatedSigningKey = response.body.signingKey as string;
```

Comment:
No need to validate these two? or just check it should not be undefined?

Reply:
Updated.